### PR TITLE
[HOLD] Helps-2438 Add filterOptions parameter to the Autocomplete component

### PR DESCRIPTION
**Pull request by BrettMugglinOlive**

### CODE REVIEW

File: ldk/javascript/src/types/oliveHelps/whisper.d.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ declare namespace WhisperService {
 
   interface Whisper {
     id: string;
-    close: Common.Readable<undefined>;
+    close: (cb: (err: Error | undefined) => void) => void;
```
Comment (MichaelOliveMcCarrick) [inline ldk/javascript/src/types/oliveHelps/whisper.d.ts:82]: Why did this get changed?

File: ldk/javascript/src/whisper/mapper.ts [hunk 2]
```diff
@@ -106,6 +106,7 @@ export function mapToInternalChildComponent(
       // eslint-disable-next-line
       const { onChange, onSelect, options } = component;
       const dataObj = { data: options };
+
```
Comment (MichaelOliveMcCarrick) [inline ldk/javascript/src/whisper/mapper.ts:109]: Extra new line

**Review by MichaelOliveMcCarrick — APPROVED** [review]

**Review by MaggieMaSS — COMMENTED** [review]
https://crosschx.atlassian.net/browse/HELPS-2447 
Seems like the Sensors&Aptitude (goja layer) ticket is not taken 🤔 And we need to wait for their implementation.

File: ldk/javascript/src/types/oliveHelps/whisper.d.ts [hunk 3]
```diff
@@ -113,7 +113,17 @@ declare namespace WhisperService {
     value: string;
   };
 
+  interface AutocompleteFilterOptions {
+    ignoreAccents?: boolean;
+    ignoreCase?: boolean;
+    limit?: number;
+    matchFrom?: 'any' | 'start';
+    stringify?: (option: AutocompleteOption) => void;
```
Comment (JosefBud-olive) [inline ldk/javascript/src/types/oliveHelps/whisper.d.ts:121]: Based on the component's code, I think stringify should be required here, or made optional in the UI

```js
  const parsedFilterOptions = filterOptions ? createFilterOptions({
    ...filterOptions,
    stringify: option => filterOptions.stringify(option),
  }) : undefined;
```

File: ldk/javascript/src/types/oliveHelps/whisper.d.ts (hunk 1 above)
Comment (JosefBud-olive) [reply ldk/javascript/src/types/oliveHelps/whisper.d.ts:82]: Also wondering this? Is this possibly leftover from when you were testing different ways to type the stringify function?

**Review by JosefBud-olive — CHANGES_REQUESTED** [review]
`stringify` not being required on our end, or leaving it required in the UI, will definitely cause some issues for folks so we should figure out what we want to do there

**Pull request closed without merging**